### DISCUSSION TITLE
Wrapper to not apply theme to children

### DISCUSSION THREAD

**Discussion by philharmonie** [saadeghi/daisyui#2749] [Q&A]
I am using daisyui together with other plugins like flowbite (mostly because daisyui is missing a datepicker). Now daisyui conflicts with the flowbite theme or other themes like my custom ones. Is there a way to add a wrapper to component to not apply the daisy theme here?
Something like the `not-prose` class of tailwind's typography plugin, that disables the prose styling for the children.

**Comment by saadeghi** [comment]
What is the conflict? Can you share a code example?

**Reply by philharmonie** [reply]
In a nutshell:
```
< !-- ... -->
<head>
    <style>
        .btn {background: red;}
    </style>
    <link href="daisyui.min.css" rel="stylesheet" type="text/css" />
</head>
< !-- ... -->
<button class="btn">Button</button>
```
The button is not red.
What I am looking for would be something like:
```
<div class="no-daisyui">
    <button class="btn">Button</button>
</div>
```

**Reply by saadeghi** [reply]
CSS variables don't work that way. If parent has a CSS variable it will be inherited to all children.

But, there are many ways to handle this:
- You can unset all theme colors for a specific wrapper like this: https://play.tailwindcss.com/EClsmGc3M1?file=css
- You can add a `prefix` to all daisyUI class names: https://daisyui.com/docs/config/
- You can add a prefix to the other library class names or use a different class name if possible.
- You can move `.btn {background: red;}` after daisyUI style so your style comes after daisyUI style and overrides it.
- You can use `!important` to override specific styles.
- You can use PostCSS plugins to change or wrap specific styles automatically.

Let me know if you have a question.
